src/node-srt.cc

```diff
@@ -267,6 +267,15 @@ Napi::Value NodeSRT::SetSockOpt(const Napi::CallbackInfo& info) {
       Napi::Error::New(env, srt_getlasterror_str()).ThrowAsJavaScriptException();
       return Napi::Number::New(env, SRT_ERROR);
     }
+  } else if (info[2].IsString()) {
+      Napi::String value = info[2].As<Napi::String>();
+      int32_t optName = option;
+      const char * optValue = std::string(value).c_str();
```

Comment:
Clang compiler spits out a warning here for understandable reasons.

that std::string objects may be kicked off the stack just after c_str returns.

meaning that the mem at the pointer is freed then, and potentially the value passed not ensured to be what you expect here at all (even if this may work, you actually pass a pointer to already deallocated mem here every time).